clarity/src/vm/analysis/type_checker/v2_05/natives/mod.rs

```diff
@@ -58,9 +58,8 @@ fn check_special_list_cons(
             type_arg.type_size()?,
         )?;
     }
-    TypeSignature::parent_list_type(&typed_args)
-        .map_err(|x| x.into())
-        .map(TypeSignature::from)
+    let list_type = TypeSignature::parent_list_type(&typed_args).map_err(CheckError::from)?;
```

Comment:
Please revert this change.  You should assume that all error types are consensus-critical.